### PR TITLE
Add commands to list OVAs and download artifacts used by EKS Anywhere

### PR DESCRIPTION
**Pull request by abhay-krishna**
This PR adds a command to list the OVA URLs corresponding to the Kubernetes version of the current EKS Anywhere cluster configuration along their checksums.

Sample output for `eksctl anywhere list ovas`:
```
$ eksctl anywhere list ovas -f eks-a-test.yaml
Bottlerocket:
  SHA256: ...
  SHA512: ...
  URI: https://...
Ubuntu:
  SHA256: ...
  SHA512: ...
  URI: https://...
```
Sample output for `eksctl anywhere download artifacts`:
```bash
$ eksctl anywhere download artifacts -f bin/arnchlm-test.yaml
$ ls *.tar.gz
eks-anywhere-downloads.tar.gz
$ tar -tzf eks-anywhere-downloads.tar.gz 
eks-anywhere-downloads
eks-anywhere-downloads/bundle-release.yaml
eks-anywhere-downloads/capi-kubeadm-bootstrap
eks-anywhere-downloads/capi-kubeadm-bootstrap/bootstrap-components.yaml
eks-anywhere-downloads/capi-kubeadm-bootstrap/metadata.yaml
eks-anywhere-downloads/capi-kubeadm-control-plane
eks-anywhere-downloads/capi-kubeadm-control-plane/control-plane-components.yaml
eks-anywhere-downloads/capi-kubeadm-control-plane/metadata.yaml
eks-anywhere-downloads/cilium
eks-anywhere-downloads/cilium/cilium.yaml
...
```
Sample output for `eksctl anywhere download artifacts` (dry run):
```bash
$ eksctl anywhere download artifacts -f bin/arnchlm-test.yaml --dry-run
Found artifact: https://dev-release-prod-pdx.s3.amazonaws.com/bundle-release.yaml

Found artifact: https://dev-release-prod-pdx.s3.us-west-2.amazonaws.com/eks-a-release.yaml

Found artifact: https://distro.eks.amazonaws.com/kubernetes-1-21/kubernetes-1-21-eks-5.yaml

Found artifact: https://dev-release-prod-pdx.s3.amazonaws.com/artifacts/v0.0.0-dev-build.352/cluster-api/manifests/bootstrap-kubeadm/v0.3.23/bootstrap-components.yaml

Found artifact: https://dev-release-prod-pdx.s3.amazonaws.com/artifacts/v0.0.0-dev-build.352/cilium/manifests/cilium/v1.9.10-eksa.1/cilium.yaml

Found artifact: https://dev-release-prod-pdx.s3.amazonaws.com/artifacts/v0.0.0-dev-build.352/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v0.1.2-rc1/bootstrap-components.yaml

Found artifact: https://dev-release-prod-pdx.s3.amazonaws.com/artifacts/v0.0.0-dev-build.352/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v0.1.2-rc1/metadata.yaml
...
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/544#" title="Author self-approved">abhay-krishna</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhay-krishna]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cmd/eksctl-anywhere/cmd/list.go
Comment (abhay-krishna) [inline cmd/eksctl-anywhere/cmd/list.go:1]: Added this to get the jobs passing, but will rebase after Rajashree adds it in her PR.

**Comment by mrajashree** [comment]
https://github.com/aws/eks-anywhere/issues/477

File: cmd/eksctl-anywhere/cmd/downloadartifacts.go
Comment (abhay-krishna) [inline cmd/eksctl-anywhere/cmd/downloadartifacts.go:153]: Should probably also support embedded manifest files

Comment (abhay-krishna) [reply cmd/eksctl-anywhere/cmd/downloadartifacts.go:153]: Probably using the [methods](https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/spec.go#L353) defined in spec.go

File: cmd/eksctl-anywhere/cmd/downloadartifacts.go
Comment (mrajashree) [inline cmd/eksctl-anywhere/cmd/downloadartifacts.go:142]: typo?
```suggestion
	if err := os.MkdirAll(filepath.Dir(filePath), 0755); err != nil {
```

Comment (abhay-krishna) [reply cmd/eksctl-anywhere/cmd/downloadartifacts.go:142]: `0o` is valid octal representation for file permissions mode, equivalent to `0755`.

File: pkg/cluster/spec.go
Comment (abhay-krishna) [reply pkg/cluster/spec.go]: Done!

File: cmd/eksctl-anywhere/cmd/downloadartifacts.go
Comment (mrajashree) [inline cmd/eksctl-anywhere/cmd/downloadartifacts.go:162]: should we also defer close gzipWriter and tarWriter since we check for error and return on line 189

File: pkg/cluster/spec.go
Comment (mrajashree) [inline pkg/cluster/spec.go:636]: since we're initializing manifests at the start of this function can this just be:
```suggestion
	manifests["cluster-api-provider-aws"] = []v1alpha1.Manifest{
		vb.Aws.Components,
		vb.Aws.ClusterTemplate,
		vb.Aws.Metadata,
	}
```
same suggestion for rest of the components below

Comment (abhay-krishna) [reply pkg/cluster/spec.go:636]: Yeah sure. I was torn between doing this and the append approach but I'm cool with this too.

File: cmd/eksctl-anywhere/cmd/downloadartifacts.go
Comment (abhay-krishna) [reply cmd/eksctl-anywhere/cmd/downloadartifacts.go:162]: Yup, good call. Thanks!

**Review by mrajashree — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/544#" title="Author self-approved">abhay-krishna</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/544#pullrequestreview-795949418" title="LGTM">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhay-krishna,mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
